cairo/tests/ef_tests/helpers/load_state_tests.py

```diff
@@ -20,39 +20,49 @@
 from ethereum_rlp.exceptions import RLPException
 from ethereum_spec_tools.evm_tools.loaders.fixture_loader import Load
 from ethereum_types.bytes import Bytes
-from ethereum_types.numeric import U64, U256
+from ethereum_types.numeric import U64
+
+from mpt.ethereum_tries import EMPTY_BYTES_HASH
+from utils.fixture_loader import LoadKethFixture
 
 
 def prepare_state_and_code_hashes(
     state: State,
 ) -> Tuple[State, Dict[Tuple[int, int], Bytes]]:
     code_hashes = {}
     for address in state._storage_tries:
-        state._storage_tries[address]._data = defaultdict(
-            lambda: defaultdict(lambda: U256(0)), state._storage_tries[address]._data
-        )
-        storage_root = compute_root(state._storage_tries[address])
+        # Map codehash to code
         account = trie_get(state._main_trie, address)
-        account = Account(
-            balance=account.balance,
-            nonce=account.nonce,
-            code=account.code,
-            storage_root=storage_root,
-            code_hash=account.code_hash,
-        )
-        code_hash_int = int.from_bytes(account.code_hash, "little")
+        account_code_hash = account.code_hash if account else EMPTY_BYTES_HASH
+        account_code = account.code if account else b""
+        code_hash_int = int.from_bytes(account_code_hash, "little")
         code_hash_low = code_hash_int & 2**128 - 1
         code_hash_high = code_hash_int >> 128
-        code_hashes[(code_hash_low, code_hash_high)] = account.code
-        trie_set(state._main_trie, address, account)
-    state._main_trie._data = defaultdict(lambda: None, state._main_trie._data)
+        code_hashes[(code_hash_low, code_hash_high)] = account_code
+
+        # Compute the storage root of the account
+        storage_root = compute_root(state._storage_tries[address])
+
+        # An account might be missing in the pre-state if it's created in the transaction, or is empty.
+        updated_account = (
+            EMPTY_ACCOUNT
+            if not account
+            else Account(
+                balance=account.balance,
+                nonce=account.nonce,
+                code=account.code,
+                storage_root=storage_root,
+                code_hash=account.code_hash,
+            )
+        )
+
+        trie_set(state._main_trie, address, updated_account)
 
     for snap in state._snapshots:
         for address in snap._storage_tries:
             snap._storage_tries[address]._data = defaultdict(
-                lambda: defaultdict(lambda: U256(0)), snap._storage_tries[address]._data
+                lambda: snap._storage_tries[address]._data
```

Comment:
For each address in the snapshot storage tries, 
Replace that address' storage trie's dictionary by a default dictionary that returns the dictionary at that address in case of key miss?

Sounds recursive / wrong

Reply:
The dimension of snap._storage_tries[address]._data is Mapping[Bytes32, U256]